control/control_debug_tools/scripts/consistency_checker.py

```diff
@@ -17,146 +17,206 @@
 
 import argparse
 import math
+import operator
+import sys
 
 from ament_index_python.packages import get_package_share_directory
 import yaml
 
 
-def read_yaml(file_path):
-    """Read YAML file and return the data."""
-    with open(file_path, "r") as file:
-        return yaml.safe_load(file)
+# Indicate the source of a parameter
+class Source:
+    MPC = 0
+    PID = 1
+    SIM = 2
+    GATE = 3
+
+    def to_string(source):
+        match source:
+            case Source.MPC:
+                return "MPC"
+            case Source.PID:
+                return "PID"
+            case Source.SIM:
+                return "Simulator model"
+            case Source.GATE:
+                return "Vehicle command gate"
+            case _:
+                return "Unknown source"
+
+
+# Condition between 2 parameters
+class Condition:
+    def __init__(self, param1, source1, param2, source2, fail_msg, op):
+        self.param1 = param1
+        self.param2 = param2
+        self.source1 = source1
+        self.source2 = source2
+        self.fail_msg = fail_msg
+        self.op = op
+
+    def evaluate(self, sources):
+        val1 = sources[self.source1][self.param1]
+        val2 = sources[self.source2][self.param2]
+        if not self.op(val1, val2):
+            return "{}[{}] = {} {} {}[{}] = {}".format(
+                Source.to_string(self.source1),
+                self.param1,
+                val1,
+                self.fail_msg,
+                Source.to_string(self.source2),
+                self.param2,
+                val2,
+            )
+        return None
 
 
-parser = argparse.ArgumentParser(
-    description="Process the parameters of the controllers and simulator."
-)
+def read_yaml(file_path):
+    """Read YAML file and return the data."""
+    try:
+        with open(file_path, "r") as file:
+            return yaml.safe_load(file)
+    except Exception as e:
+        print(e)
+        sys.exit(-1)
 
-parser.add_argument(
-    "--vehicle_description",
-    help="Specify the vehicle description package name",
-    default="sample_vehicle_description",
-)
-parser.add_argument("--mpc_param_file", help="Override the default MPC parameter file path")
-parser.add_argument("--pid_param_file", help="Override the default PID parameter file path")
-parser.add_argument(
-    "--simulator_model_param_file", help="Override the default simulator model parameter file path"
-)
 
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Process the parameters of the controllers and simulator."
+    )
 
-def main():
+    parser.add_argument(
+        "--vehicle_description",
+        help="Specify the vehicle description package name",
+        default="sample_vehicle_description",
```

Comment:
@maxime-clem 
prefix of `autoware_` is better to be added?
I got a following error with original implementation
```
"package 'sample_vehicle_description' not found, searching:
```

Reply:
On the current AWF repo we use `sample_vehicle_description`. I don't think `autoware_sample_vehicle_description` exists so I will probably revert the change.

Reply:
Oh sorry I didn't know that